app/models/manageiq/providers/openstack/cloud_manager/vm.rb

```diff
@@ -1,12 +1,10 @@
 class ManageIQ::Providers::Openstack::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
-  include_concern 'ManageIQ::Providers::Openstack::CloudManager::VmOrTemplateShared'
-
-  include_concern 'Operations'
-  include_concern 'RemoteConsole'
-  include_concern 'Resize'
-  include_concern 'AssociateIp'
-  include_concern 'ManageSecurityGroups'
-
+  include ManageIQ::Providers::Openstack::CloudManager::VmOrTemplateShared
```

Comment:
It's interesting to point out that VmOrTemplateShared is NOT within the Vm namespace 
so you can't do either of the options below because it's looking for them relative to the current namespace:

### include VmOrTemplateShared
 
```diff --git a/app/models/manageiq/providers/openstack/cloud_manager/vm.rb b/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
index c585995c..4dafdfbc 100644
--- a/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
+++ b/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
@@ -1,5 +1,5 @@
 class ManageIQ::Providers::Openstack::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
-  include ManageIQ::Providers::Openstack::CloudManager::VmOrTemplateShared
+  include VmOrTemplateShared
   include Operations
   include RemoteConsole
   include Resize
```

```
Failure/Error: include VmOrTemplateShared

NameError:
  uninitialized constant ManageIQ::Providers::Openstack::CloudManager::Vm::VmOrTemplateShared
# ./app/models/manageiq/providers/openstack/cloud_manager/vm.rb:2:in `<class:Vm>'
# ./app/models/manageiq/providers/openstack/cloud_manager/vm.rb:1:in `<main>'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:287:in `const_get'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:287:in `block in constantize'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:283:in `each'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:283:in `inject'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:283:in `constantize'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/inflector/methods.rb:329:in `safe_constantize'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/activesupport-6.1.7.6/lib/active_support/core_ext/string/inflections.rb:87:in `safe_constantize'
# /Users/joerafaniello/Code/manageiq/lib/extensions/descendant_loader.rb:248:in `load_subclasses'
# /Users/joerafaniello/Code/manageiq/lib/extensions/descendant_loader.rb:268:in `descendants'
# ./spec/manageiq/spec/automated_review/ar_models_must_have_tables_spec.rb:2:in `block in <main>'
# ./spec/manageiq/spec/automated_review/ar_models_must_have_tables_spec.rb:1:in `<main>'

```

### Or include CloudManager::VmOrTemplateShared

```diff --git a/app/models/manageiq/providers/openstack/cloud_manager/vm.rb b/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
index c585995c..6c659aaa 100644
--- a/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
+++ b/app/models/manageiq/providers/openstack/cloud_manager/vm.rb
@@ -1,5 +1,5 @@
 class ManageIQ::Providers::Openstack::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
-  include ManageIQ::Providers::Openstack::CloudManager::VmOrTemplateShared
+  include CloudManager::VmOrTemplateShared
```


```
An error occurred while loading ./spec/models/manageiq/providers/openstack/cloud_manager/vm_spec.rb.
Failure/Error: include CloudManager::VmOrTemplateShared

NameError:
  uninitialized constant ManageIQ::Providers::Openstack::CloudManager::Vm::CloudManager
# ./app/models/manageiq/providers/openstack/cloud_manager/vm.rb:2:in `<class:Vm>'
# ./app/models/manageiq/providers/openstack/cloud_manager/vm.rb:1:in `<main>'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
# /Users/joerafaniello/.gem/ruby/3.0.6/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
# ./spec/models/manageiq/providers/openstack/cloud_manager/vm_spec.rb:1:in `<main>'

Reply:
I guess you could do 
```suggestion
  include parent::VmOrTemplateShared
```

Reply:
Yeah that makes sense, if it were in the `Vm` namespace then we'd have to do the full path for `Openstack::CloudManager::Template` or vice versa

Reply:
> include ManageIQ::Providers::Openstack::CloudManager::VmOrTemplateShared
>  include module_parent::VmOrTemplateShared

I'm leaning towards the explicit fully qualified constant path.  It's far more obvious it's "different".